core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -157,10 +157,24 @@ public static State fromString(String state) {
         }
     }
 
+    static {
+        final int maxNumShards = Integer.parseInt(System.getProperty("index.max_number_of_shards", "1024"));
```

Comment:
I've been trying to prefix the few system properties we have with `es`; I wonder if we should do this here?